cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternals.qll

```diff
@@ -719,14 +711,19 @@ predicate ssaFlow(Node nodeFrom, Node nodeTo) {
   )
 }
 
+private predicate isArgumentOfCallable(DataFlowCall call, ArgumentNode arg) {
+  arg.argumentOf(call, _)
+}
+
+/** Holds if there is def-use or use-use flow from `pun` to `nodeTo`. */
 predicate postUpdateFlow(PostUpdateNode pun, Node nodeTo) {
-  exists(Node preUpdate, Node nFrom, boolean uncertain, SsaDefOrUse defOrUse |
+  exists(UseOrPhi use, Node preUpdate |
+    adjustForPointerArith(pun, use) and
+    useToNode(use, nodeTo) and
     preUpdate = pun.getPreUpdateNode() and
-    ssaFlowImpl(defOrUse, nFrom, nodeTo, uncertain)
-  |
-    if uncertain = true
-    then preUpdate = [nFrom, getAPriorDefinition(defOrUse)]
-    else preUpdate = nFrom
```

Comment:
Why do we no longer need this case distinction?

Reply:
Because I don't think `getAPriorDefinition(defOrUse)` ever returns a `PostUpdateNode`. It ends up getting its result from `nodeToDefOrUse`, and in both of those cases we get an `IndirectOperand` or an `IndirectInstruction`, and a `PostUpdateNode` is always an instance of the `TPostFieldUpdateNode` branch (when it's a write to a field) or the `TIndirectArgumentOutNode` branch (when it's the output node after writing to an argument).